include/trainer_info.h

```diff
@@ -19,39 +19,291 @@ typedef struct TrainerInfo {
+
+/**
+ * @brief Set the trainer's name.
+ * 
+ * @param info  The trainer info.
+ * @param name  Player's new name, as a Strbuf struct.
+ */
+void TrainerInfo_SetNameFromStrbuf(TrainerInfo *info, const Strbuf *name);
+
+/**
+ * @brief Get the trainer's name.
+ * 
+ * @param info  The trainer info.
+ * @return The trainer's name, as a character buffer.
+ */
+const charcode_t* TrainerInfo_Name(const TrainerInfo *info);
+
+/**
+ * @brief Get the trainer's name.
+ * 
+ * @param info  The trainer info.
+ * @param name  Pre-allocated Strbuf struct to receive the trainer's name.
+ */
+void TrainerInfo_NameStrbuf(const TrainerInfo *info, Strbuf *name);
+
+/**
+ * @brief Get the trainer's name.
+ * 
+ * @param info      The trainer info.
+ * @param heapID    Heap on which to allocate the new Strbuf struct.
+ * @return The trainer's name, as a Strbuf struct.
+ */
+Strbuf* TrainerInfo_NameNewStrbuf(const TrainerInfo *info, int heapID);
+
+/**
+ * @brief Set the trainer's ID.
+ * 
+ * @param info  The trainer info.
+ * @param id    New trainer ID.
+ */
+void TrainerInfo_SetID(TrainerInfo *info, u32 id);
+
+/**
+ * @brief Get the trainer's ID.
+ * 
+ * Note that this is the complete 32-byte ID; it includes both the visible
+ * trainer ID and the invisible "secret ID."
+ * 
+ * @param info  The trainer info.
+ * @return The trainer's ID.
+ */
+u32 TrainerInfo_ID(const TrainerInfo *info);
+
+/**
+ * @brief Get the lower half of the trainer's ID.
+ * 
+ * This constitutes only the visible trainer ID.
+ * 
+ * @param info  The trainer info.
+ * @return The lower half of the trainer's ID.
+ */
+u16 TrainerInfo_ID_LowHalf(const TrainerInfo *info);
+
+/**
+ * @brief Set the trainer's gender.
+ * 
+ * @param info      The trainer info.
+ * @param gender    New trainer gender.
+ */
+void TrainerInfo_SetGender(TrainerInfo *info, int gender);
+
+/**
+ * @brief Get the trainer's gender.
+ * 
+ * @param info  The trainer info.
+ * @return The trainer's gender.
+ */
+u32 TrainerInfo_Gender(const TrainerInfo *info);
+
+/**
+ * @brief Check if the trainer has a certain badge.
+ * 
+ * @param info      The trainer info.
+ * @param badgeNum  Number of the badge to check, 0-indexed.
+ * @return TRUE if the trainer has the badge, FALSE otherwise.
+ */
+BOOL TrainerInfo_HasBadge(const TrainerInfo *info, int badgeNum);
+
+/**
+ * @brief Set the flag denoting possession of a given badge.
+ * 
+ * @param info      The trainer info.
+ * @param badgeNum  Number of the badge to set.
+ */
+void TrainerInfo_SetBadge(TrainerInfo *info, int badgeNum);
+
+/**
+ * @brief Get the number of badges owned by the trainer.
+ * 
+ * @param info  The trainer info.
+ * @return Number of badges acquired.
+ */
+int TrainerInfo_BadgeCount(const TrainerInfo *info);
+
+/**
+ * @brief Get the trainer's current money.
+ * 
+ * @param info  The trainer info.
+ * @return How much money the trainer has.
+ */
+u32 TrainerInfo_Money(const TrainerInfo *info);
```

Comment:
`TrainerInfo_GetMoney`

Reply:
Resolving in favor of any further discussion on the above comment.